storage/factory/pruningStorerFactory.go

```diff
@@ -0,0 +1,334 @@
+package factory
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/pruning"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("storage/factory")
+
+// StorageServiceFactory handles the creation of storage services for both meta and shards
+type StorageServiceFactory struct {
+	generalConfig      *config.Config
+	shardCoordinator   sharding.Coordinator
+	uniqueID           string
+	epochStartNotifier storage.EpochStartNotifier
+}
+
+// NewStorageServiceFactory will return a new instance of StorageServiceFactory
+func NewStorageServiceFactory(
+	config *config.Config,
+	shardCoordinator sharding.Coordinator,
+	uniqueID string,
+	epochStartNotifier storage.EpochStartNotifier,
+) (*StorageServiceFactory, error) {
+	if config == nil {
```

Comment:
the validity of the config can be inside the factory. not in main.go. like checking epochtokeep and numpersisters.

Reply:
done